redwood-layout-composeui/src/commonMain/kotlin/app/cash/redwood/layout/composeui/ComposeUiBox.kt

```diff
@@ -0,0 +1,122 @@
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.BiasAlignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import app.cash.redwood.Modifier as RedwoodModifier
+import app.cash.redwood.layout.api.Constraint
+import app.cash.redwood.layout.api.CrossAxisAlignment
+import app.cash.redwood.layout.widget.Box
+import app.cash.redwood.ui.Margin
+import app.cash.redwood.widget.compose.ComposeWidgetChildren
+
+internal class ComposeUiBox(
+  private val backgroundColor: Int? = null,
+) : Box<@Composable () -> Unit> {
+  override val children = ComposeWidgetChildren()
+
+  private var width by mutableStateOf(Constraint.Wrap)
+  private var height by mutableStateOf(Constraint.Wrap)
+  private var margin by mutableStateOf(Margin.Zero)
+  private var alignment by mutableStateOf(BiasAlignment(-1f, -1f))
+
+  override val value = @Composable {
+    Box(computeModifier(), contentAlignment = alignment) {
+      children.Render()
+    }
+  }
+
+  @Composable
+  private fun computeModifier(): Modifier {
+    var modifier: Modifier = Modifier
+    modifier = if (width == Constraint.Fill) {
+      modifier.fillMaxWidth()
+    } else {
+      modifier.wrapContentWidth(Alignment.Start, unbounded = true)
+    }
+    modifier = if (height == Constraint.Fill) {
+      modifier.fillMaxHeight()
+    } else {
+      modifier.wrapContentHeight(Alignment.Top, unbounded = true)
+    }
+    modifier = margin.let { margin ->
+      modifier.padding(
+        start = margin.start.toDp(),
+        top = margin.top.toDp(),
+        end = margin.end.toDp(),
+        bottom = margin.bottom.toDp(),
+      )
+    }
+    backgroundColor?.let { backgroundColor ->
+      modifier = modifier.background(Color(backgroundColor))
+    }
+    return modifier
+  }
+
+  override var modifier: RedwoodModifier = RedwoodModifier
+
+  override fun width(width: Constraint) {
+    this.width = width
+  }
+
+  override fun height(height: Constraint) {
+    this.height = height
+  }
+
+  override fun margin(margin: Margin) {
+    this.margin = margin
+  }
+
+  override fun horizontalAlignment(horizontalAlignment: CrossAxisAlignment) {
+    alignment = BiasAlignment(
+      horizontalBias = when (horizontalAlignment) {
+        CrossAxisAlignment.Start -> -1f
+        CrossAxisAlignment.Center -> 0f
+        CrossAxisAlignment.End -> 1f
+        CrossAxisAlignment.Stretch -> TODO()
```

Comment:
No idea how to do this. I think we need `BoxWithConstraints` or just a straight-up `Layout`. Future problems.

Reply:
We can partly accomplish this using `BoxScope.matchParentSize`. However, I don't think we can't match on one dimension - only both.

I took a quick look at the implementation and it doesn't look like we can easily wrap the `Modifier` to only wrap one dimension unfortunately. I think we probably do need a custom `Layout` implementation where we copy `Box`'s code and modify the `MeasurePolicy`.